lib/prompts/checkbox.js

```diff
@@ -86,7 +88,7 @@ Prompt.prototype.render = function (error) {
   var message = this.getQuestion();
   var bottomContent = '';
 
-  if (!this.spaceKeyPressed) {
+  if (this.hints) {
```

Comment:
Instead of tracking the `this.hints` value in many methods, it'd be simpler to think about `firstRender`.

What the code you changed basically means only render the hints on the initial render. So I think that's how we should model the code. Render hints on first render, hide afterwards.